<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PreferencesDialog</class>
 <widget class="QDialog" name="PreferencesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>414</width>
    <height>379</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="streamingTab">
      <attribute name="title">
       <string>Streaming</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="1" column="0">
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="automaticRGBCheckBox">
         <property name="text">
          <string>Automatically set RGB LEDs when connecting/disconnecting</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="ledsLabel">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>RGB LEDs</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="2">
        <spacer name="verticalSpacer_9">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>369</width>
           <height>243</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QLabel" name="downsamplingLabel">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Downsampling</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QCheckBox" name="downsampleCheckBox">
         <property name="text">
          <string>Downsample high speed channels (for time domain graph only)</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="unitTab">
      <attribute name="title">
       <string>Units</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="unitsLabel">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Display Units</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>13</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="4" column="0" colspan="2">
        <spacer name="verticalSpacer_7">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1">
        <widget class="UnitPreferencesWidget" name="unitPreferences" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="outputTab">
      <attribute name="title">
       <string>Output</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2" columnstretch="0,1,0">
       <item row="6" column="0" colspan="3">
        <spacer name="verticalSpacer_8">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>369</width>
           <height>237</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0" colspan="3">
        <widget class="QLabel" name="outputLocationLabel">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Output Location</string>
         </property>
        </widget>
       </item>
       <item row="5" column="2">
        <widget class="QSpinBox" name="compressionLevel">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>9</number>
         </property>
         <property name="value">
          <number>6</number>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="3">
        <widget class="QLabel" name="compressionLabel">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Compression</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QLabel" name="compressionLevelLabel">
         <property name="text">
          <string>Compression Level (default 6)</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1" colspan="2">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLineEdit" name="outputLocation"/>
         </item>
         <item>
          <widget class="QPushButton" name="browseButton">
           <property name="text">
            <string>Browse...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="uploadTab">
      <attribute name="title">
       <string>Cloud Upload</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="5" column="3">
        <widget class="QLineEdit" name="uploadAccessKeyID">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="5" column="2">
        <widget class="QLabel" name="uploadAccessKeyIDLabel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>AWS Access Key ID</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <spacer name="horizontalSpacer_5">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>13</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1" colspan="3">
        <widget class="QCheckBox" name="enableUpload">
         <property name="text">
          <string>Enable AWS S3 Upload</string>
         </property>
        </widget>
       </item>
       <item row="6" column="3">
        <widget class="QLineEdit" name="uploadSecretAccessKey">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="echoMode">
          <enum>QLineEdit::PasswordEchoOnEdit</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="s3BucketLabel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>S3 Bucket</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="3">
        <widget class="QLabel" name="uploadLabel">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Amazon S3 Upload</string>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QLabel" name="uploadDirectoryLabel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Upload Directory</string>
         </property>
        </widget>
       </item>
       <item row="4" column="3">
        <widget class="QLineEdit" name="uploadDirectory">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>13</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="8" column="0" colspan="4">
        <spacer name="verticalSpacer_10">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>369</width>
           <height>116</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="3">
        <widget class="QLineEdit" name="s3Bucket">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="7" column="2" colspan="2">
        <widget class="QCheckBox" name="uploadDeleteOriginal">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Delete original .apd after successful upload</string>
         </property>
        </widget>
       </item>
       <item row="6" column="2">
        <widget class="QLabel" name="uploadSecretAccessKeyLabel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>AWS Secret Access Key</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QLabel" name="awsRegionLabel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>S3 Bucket AWS Region</string>
         </property>
        </widget>
       </item>
       <item row="3" column="3">
        <widget class="QLineEdit" name="awsRegion">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>UnitPreferencesWidget</class>
   <extends>QWidget</extends>
   <header location="global">hyperborea.unit_preferences</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>automaticRGBCheckBox</tabstop>
  <tabstop>outputLocation</tabstop>
  <tabstop>browseButton</tabstop>
  <tabstop>compressionLevel</tabstop>
  <tabstop>enableUpload</tabstop>
  <tabstop>s3Bucket</tabstop>
  <tabstop>awsRegion</tabstop>
  <tabstop>uploadDirectory</tabstop>
  <tabstop>uploadAccessKeyID</tabstop>
  <tabstop>uploadSecretAccessKey</tabstop>
  <tabstop>uploadDeleteOriginal</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>369</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>369</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>s3BucketLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>113</x>
     <y>93</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>s3Bucket</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>231</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>235</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>uploadDirectoryLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>168</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>uploadDirectory</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>192</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>uploadAccessKeyIDLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>185</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>uploadAccessKeyID</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>341</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>391</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>uploadSecretAccessKeyLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>171</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>uploadSecretAccessKey</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>258</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>375</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>uploadDeleteOriginal</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>279</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>awsRegionLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>120</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableUpload</sender>
   <signal>toggled(bool)</signal>
   <receiver>awsRegion</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>277</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>271</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
